client/lib/plugins/utils.js

```diff
@@ -164,6 +164,9 @@ export function normalizePluginData( plugin, pluginData ) {
 			case 'sections':
 				const cleanItem = {};
 				for ( const sectionKey of Object.keys( item ) ) {
+					if ( ! item[ sectionKey ] ) {
+						throw new Error( 'Section expected for key "' + sectionKey + '"' );
```

Comment:
Perhaps:
```jsx
`Section expected for key "${ sectionKey }"`
```

Reply:
Yeah, I thought about that too. I'll update it.

Reply:
why is this possible? it looks like every `item[ sectionKey ]` _must_ exist because of the `for…of` loop otherwise it wouldn't be set. also, is `throw` appropriate? it sounds like maybe you are moving the error earlier before it causes problems deeper in the app, in which case I thank you. otherwise if this causes Calypso to crash then could we do something differently?

also, if this comes via the API, should we look at handling it in the data layer when it arrives?

Reply:
Yes, it's already erroring. This is just to get it to error with better information.